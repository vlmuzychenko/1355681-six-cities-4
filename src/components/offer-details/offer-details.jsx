import React from "react";
import PropTypes from "prop-types";
import ReviewsList from "../reviews-list/reviews-list.jsx";
import ReviewForm from "../review-form/review-form.jsx";
import NearOffersList from "../near-offers-list/near-offers-list.jsx";
import Map from "../map/map.jsx";
import withHandleForm from "../../hocs/with-handle-form/with-handle-form.js";
import {getRatingInPercent, getPluralizedString} from "../../utils/common.js";
import {connect} from "react-redux";
import {Operation as ReviewsOperation} from "../../reducer/reviews/reviews.js";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import {getCurrentCity} from "../../reducer/data/selectors.js";
import {getCurrentOffer} from "../../reducer/main/selectors.js";
import {getCurrentReviews} from "../../reducer/reviews/selectors.js";
import {getNearOffers} from "../../reducer/nearby/selectors.js";
import {OfferTypes} from "../../const.js";

const ReviewFormWithHandler = withHandleForm(ReviewForm);

const OfferDetails = (props) => {
  const {authorizationStatus, currentOffer, currentCity, currentReviews, nearOffers, onSubmit} = props;
  const {id, title, price, images, description, rating, isPremium, type, bedrooms, maxAdults, goods, host} = currentOffer;
  const premiumMark = isPremium ? <div className="property__mark"><span>Premium</span></div> : ``;

  return (
    <React.Fragment>
      <div style={{display: `none`}}>
        <svg xmlns="http://www.w3.org/2000/svg"><symbol id="icon-arrow-select" viewBox="0 0 7 4"><path fillRule="evenodd" clipRule="evenodd" d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"></path></symbol><symbol id="icon-bookmark" viewBox="0 0 17 18"><path d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"></path></symbol><symbol id="icon-star" viewBox="0 0 13 12"><path fillRule="evenodd" clipRule="evenodd" d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"></path></symbol></svg>
      </div>

      <div className="page">
        {props.children}

        <main className="page__main page__main--property">
          <section className="property">
            <div className="property__gallery-container container">
              <div className="property__gallery">
                {images.map((image, index) => {
                  return (
                    <div className="property__image-wrapper" key={image + index}>
                      <img className="property__image" src={image} alt="Photo studio" />
                    </div>
                  );
                })}
              </div>
            </div>
            <div className="property__container container">
              <div className="property__wrapper">
                {premiumMark}
                <div className="property__name-wrapper">
                  <h1 className="property__name">
                    {title}
                  </h1>
                  <button className="property__bookmark-button button" type="button">
                    <svg className="property__bookmark-icon" width="31" height="33">
                      <use xlinkHref="#icon-bookmark"></use>
                    </svg>
                    <span className="visually-hidden">To bookmarks</span>
                  </button>
                </div>
                <div className="property__rating rating">
                  <div className="property__stars rating__stars">
                    <span style={{width: getRatingInPercent(rating)}}></span>
                    <span className="visually-hidden">Rating</span>
                  </div>
                  <span className="property__rating-value rating__value">{rating}</span>
                </div>
                <ul className="property__features">
                  <li className="property__feature property__feature--entire">
                    {OfferTypes.get(type)}
                  </li>
                  <li className="property__feature property__feature--bedrooms">
                    {bedrooms} {getPluralizedString(bedrooms, `Bedroom`)}
                  </li>
                  <li className="property__feature property__feature--adults">
                    Max {maxAdults} {getPluralizedString(maxAdults, `adult`)}
                  </li>
                </ul>
                <div className="property__price">
                  <b className="property__price-value">&euro;{price}</b>
                  <span className="property__price-text">&nbsp;night</span>
                </div>
                <div className="property__inside">
                  <h2 className="property__inside-title">What&apos;s inside</h2>
                  <ul className="property__inside-list">
                    {goods.map((good, index) => {
                      return (
                        <li className="property__inside-item" key={good + index}>
                          {good}
                        </li>
                      );
                    })}
                  </ul>
                </div>
                <div className="property__host">
                  <h2 className="property__host-title">Meet the host</h2>
                  <div className="property__host-user user">
                    <div className={`property__avatar-wrapper user__avatar-wrapper ${host.super ? `property__avatar-wrapper--pro` : `` }`}>
                      <img className="property__avatar user__avatar" src={host.avatarUrl} width="74" height="74" alt="Host avatar" />
                    </div>
                    <span className="property__user-name">
                      {host.name}
                    </span>
                  </div>
                  <div className="property__description">
                    <p className="property__text">
                      {description}
                    </p>
                  </div>
                </div>
                <section className="property__reviews reviews">
                  <h2 className="reviews__title">{getPluralizedString(currentReviews.length, `Review`)} &middot; <span className="reviews__amount">{currentReviews.length}</span></h2>
                  <ReviewsList
                    reviews={currentReviews}
                  />
                  {authorizationStatus === AuthorizationStatus.AUTH ?
                    <ReviewFormWithHandler
                      offerId={id}
                      onSubmit={onSubmit} />
                    : ``}
                </section>
              </div>
            </div>
            <section className="property__map map">
              <Map
                currentOffer={currentOffer}
                currentCity={currentCity}
                offers={nearOffers}
              />
            </section>
          </section>
          <div className="container">
            <section className="near-places places">
              <h2 className="near-places__title">Other places in the neighbourhood</h2>
              <NearOffersList
                offers={nearOffers}
              />
            </section>
          </div>
        </main>
      </div>
    </React.Fragment>
  );
};

OfferDetails.propTypes = {
  children: PropTypes.element,
  authorizationStatus: PropTypes.string.isRequired,
  currentOffer: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    previewImage: PropTypes.string.isRequired,
    images: PropTypes.arrayOf(PropTypes.string).isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    isPremium: PropTypes.bool.isRequired,
    type: PropTypes.string.isRequired,
    bedrooms: PropTypes.number.isRequired,
    maxAdults: PropTypes.number.isRequired,
    goods: PropTypes.arrayOf(PropTypes.string).isRequired,
    host: PropTypes.shape({
      name: PropTypes.string.isRequired,
      super: PropTypes.bool.isRequired,
      avatarUrl: PropTypes.string.isRequired,
    }).isRequired,
    city: PropTypes.shape({
      name: PropTypes.string.isRequired,
      coords: PropTypes.arrayOf(PropTypes.number).isRequired,
    }).isRequired,
    coords: PropTypes.arrayOf(PropTypes.number).isRequired,
  }),
  nearOffers: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    previewImage: PropTypes.string.isRequired,
    images: PropTypes.arrayOf(PropTypes.string).isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    isPremium: PropTypes.bool.isRequired,
    type: PropTypes.string.isRequired,
    bedrooms: PropTypes.number.isRequired,
    maxAdults: PropTypes.number.isRequired,
    goods: PropTypes.arrayOf(PropTypes.string).isRequired,
    host: PropTypes.shape({
      name: PropTypes.string.isRequired,
      super: PropTypes.bool.isRequired,
      avatarUrl: PropTypes.string.isRequired,
    }).isRequired,
    city: PropTypes.shape({
      name: PropTypes.string.isRequired,
      coords: PropTypes.arrayOf(PropTypes.number).isRequired,
    }).isRequired,
    coords: PropTypes.arrayOf(PropTypes.number).isRequired,
  })),
  currentCity: PropTypes.shape({
    name: PropTypes.string,
    coords: PropTypes.arrayOf(PropTypes.number),
    zoom: PropTypes.number,
  }).isRequired,
  currentReviews: PropTypes.any,
  onSubmit: PropTypes.func,
};

const mapStateToProps = (state) => {
  return {
    currentOffer: getCurrentOffer(state),
    currentReviews: getCurrentReviews(state),
    currentCity: getCurrentCity(state),
    nearOffers: getNearOffers(state),
  };
};

const mapDispatchToProps = (dispatch) => ({
  onSubmit(offerId, reviewInfo, resetForm) {
    dispatch(ReviewsOperation.sendReview(offerId, reviewInfo, resetForm));
  }
});

export {OfferDetails};
export default connect(mapStateToProps, mapDispatchToProps)(OfferDetails);
