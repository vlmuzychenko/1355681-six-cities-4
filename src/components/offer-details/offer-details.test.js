import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space.js";
import {OfferDetails} from "./offer-details.jsx";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";

const mockStore = configureStore([]);

const offerMock = {
  id: 1,
  title: `Beautiful & luxurious apartment at great location`,
  price: 120,
  previewImage: `img/apartment-01.jpg`,
  images: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
  description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
  rating: 4,
  isPremium: true,
  type: `apartment`,
  bedrooms: 4,
  maxAdults: 4,
  isFavorite: false,
  goods: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitche`, `Dishwasher`, `Cabel TV`, `Fridge`],
  host: {
    name: `Angelina`,
    super: true,
    avatarUrl: `img/avatar-angelina.jpg`,
  },
  city: {
    name: `Paris`,
    coords: [48.865, 2.35],
  },
  coords: [52.3909553943508, 4.85309666406198],
};

const reviewsMock = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 1,
    rating: 4,
    user: {
      avatarUrl: `img/1.png`,
      id: 4,
      isPro: false,
      name: `Max`
    }
  }
];

const AuthorizationStatusMock = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

it(`Should Offer Details render correctly`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      currentOffer: offerMock,
    },
    [NameSpace.REVIEWS]: {
      currentReviews: reviewsMock,
    },
    [NameSpace.NEARBY]: {
      nearOffers: [offerMock],
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatusMock.NO_AUTH,
      authorizationInfo: null,
    },
  });

  const onSubmit = jest.fn();
  const getOfferData = jest.fn();

  const tree = renderer
    .create(
        <BrowserRouter>
          <Provider store={store}>
            <OfferDetails
              id={`1`}
              authorizationStatus={AuthorizationStatusMock.NO_AUTH}
              offer={offerMock}
              nearOffers={[offerMock]}
              reviews={reviewsMock}
              onSubmit={onSubmit}
              getOfferData={getOfferData}
            />
          </Provider>
        </BrowserRouter>,
        {
          createNodeMock: () => document.createElement(`div`)
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
