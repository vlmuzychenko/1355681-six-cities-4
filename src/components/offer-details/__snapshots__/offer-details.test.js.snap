// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Offer Details render correctly 1`] = `
Array [
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clipRule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clipRule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>,
  <div
    className="page"
  >
    <header
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__left"
          >
            <a
              className="header__logo-link header__logo-link--active"
            >
              <img
                alt="6 cities logo"
                className="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            className="header__nav"
          >
            <ul
              className="header__nav-list"
            >
              <li
                className="header__nav-item user"
              >
                <a
                  className="header__nav-link header__nav-link--profile"
                  href="#"
                >
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    className="header__user-name user__name"
                  >
                    Oliver.conner@gmail.com
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      className="page__main page__main--property"
    >
      <section
        className="property"
      >
        <div
          className="property__gallery-container container"
        >
          <div
            className="property__gallery"
          >
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="img/room.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="img/apartment-01.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="img/apartment-02.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="img/apartment-03.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="img/studio-01.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="img/apartment-01.jpg"
              />
            </div>
          </div>
        </div>
        <div
          className="property__container container"
        >
          <div
            className="property__wrapper"
          >
            <div
              className="property__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="property__name-wrapper"
            >
              <h1
                className="property__name"
              >
                Beautiful & luxurious apartment at great location
              </h1>
              <button
                className="property__bookmark-button button"
                type="button"
              >
                <svg
                  className="property__bookmark-icon"
                  height="33"
                  width="31"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="property__rating rating"
            >
              <div
                className="property__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
              <span
                className="property__rating-value rating__value"
              >
                4
              </span>
            </div>
            <ul
              className="property__features"
            >
              <li
                className="property__feature property__feature--entire"
              >
                Apartment
              </li>
              <li
                className="property__feature property__feature--bedrooms"
              >
                4
                 
                Bedrooms
              </li>
              <li
                className="property__feature property__feature--adults"
              >
                Max 
                4
                 
                adults
              </li>
            </ul>
            <div
              className="property__price"
            >
              <b
                className="property__price-value"
              >
                €
                120
              </b>
              <span
                className="property__price-text"
              >
                 night
              </span>
            </div>
            <div
              className="property__inside"
            >
              <h2
                className="property__inside-title"
              >
                What's inside
              </h2>
              <ul
                className="property__inside-list"
              >
                <li
                  className="property__inside-item"
                >
                  Wi-Fi
                </li>
                <li
                  className="property__inside-item"
                >
                  Washing machine
                </li>
                <li
                  className="property__inside-item"
                >
                  Towels
                </li>
                <li
                  className="property__inside-item"
                >
                  Heating
                </li>
                <li
                  className="property__inside-item"
                >
                  Coffee machine
                </li>
                <li
                  className="property__inside-item"
                >
                  Baby seat
                </li>
                <li
                  className="property__inside-item"
                >
                  Kitche
                </li>
                <li
                  className="property__inside-item"
                >
                  Dishwasher
                </li>
                <li
                  className="property__inside-item"
                >
                  Cabel TV
                </li>
                <li
                  className="property__inside-item"
                >
                  Fridge
                </li>
              </ul>
            </div>
            <div
              className="property__host"
            >
              <h2
                className="property__host-title"
              >
                Meet the host
              </h2>
              <div
                className="property__host-user user"
              >
                <div
                  className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
                >
                  <img
                    alt="Host avatar"
                    className="property__avatar user__avatar"
                    height="74"
                    src="img/avatar-angelina.jpg"
                    width="74"
                  />
                </div>
                <span
                  className="property__user-name"
                >
                  Angelina
                </span>
              </div>
              <div
                className="property__description"
              >
                <p
                  className="property__text"
                >
                  An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.
                </p>
              </div>
            </div>
            <section
              className="property__reviews reviews"
            >
              <h2
                className="reviews__title"
              >
                Review
                 · 
                <span
                  className="reviews__amount"
                >
                  1
                </span>
              </h2>
              <ul
                className="reviews__list"
              >
                <li
                  className="reviews__item"
                >
                  <div
                    className="reviews__user user"
                  >
                    <div
                      className="reviews__avatar-wrapper user__avatar-wrapper"
                    >
                      <img
                        alt="Reviews avatar"
                        className="reviews__avatar user__avatar"
                        height="54"
                        src="img/avatar-max.jpg"
                        width="54"
                      />
                    </div>
                    <span
                      className="reviews__user-name"
                    >
                      Max
                    </span>
                  </div>
                  <div
                    className="reviews__info"
                  >
                    <div
                      className="reviews__rating rating"
                    >
                      <div
                        className="reviews__stars rating__stars"
                      >
                        <span
                          style={
                            Object {
                              "width": "80%",
                            }
                          }
                        />
                        <span
                          className="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <p
                      className="reviews__text"
                    >
                      A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
                    </p>
                    <time
                      className="reviews__time"
                      dateTime="2019-24-04"
                    >
                      April 2019
                    </time>
                  </div>
                </li>
              </ul>
              <form
                action="#"
                className="reviews__form form"
                method="post"
              >
                <label
                  className="reviews__label form__label"
                  htmlFor="review"
                >
                  Your review
                </label>
                <div
                  className="reviews__rating-form form__rating"
                >
                  <input
                    className="form__rating-input visually-hidden"
                    id="5-stars"
                    name="rating"
                    type="radio"
                    value="5"
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="5-stars"
                    title="perfect"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    className="form__rating-input visually-hidden"
                    id="4-stars"
                    name="rating"
                    type="radio"
                    value="4"
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="4-stars"
                    title="good"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    className="form__rating-input visually-hidden"
                    id="3-stars"
                    name="rating"
                    type="radio"
                    value="3"
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="3-stars"
                    title="not bad"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    className="form__rating-input visually-hidden"
                    id="2-stars"
                    name="rating"
                    type="radio"
                    value="2"
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="2-stars"
                    title="badly"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    className="form__rating-input visually-hidden"
                    id="1-star"
                    name="rating"
                    type="radio"
                    value="1"
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="1-star"
                    title="terribly"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                </div>
                <textarea
                  className="reviews__textarea form__textarea"
                  id="review"
                  name="review"
                  placeholder="Tell how was your stay, what you like and what can be improved"
                />
                <div
                  className="reviews__button-wrapper"
                >
                  <p
                    className="reviews__help"
                  >
                    To submit review please make sure to set 
                    <span
                      className="reviews__star"
                    >
                      rating
                    </span>
                     and describe your stay with at least 
                    <b
                      className="reviews__text-amount"
                    >
                      50 characters
                    </b>
                    .
                  </p>
                  <button
                    className="reviews__submit form__submit button"
                    disabled=""
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </form>
            </section>
          </div>
        </div>
        <section
          className="property__map map"
        >
          <div
            id="map"
          />
        </section>
      </section>
      <div
        className="container"
      >
        <section
          className="near-places places"
        >
          <h2
            className="near-places__title"
          >
            Other places in the neighbourhood
          </h2>
          <div
            className="places__list near-places__list "
          >
            <article
              className="place-card near-places__card "
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              
              <div
                className="place-card__image-wrapper near-places__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/room.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      80
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/offer"
                    onClick={[Function]}
                  >
                    Wood and stone place 1
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Private Room
                </p>
              </div>
            </article>
            <article
              className="place-card near-places__card "
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              
              <div
                className="place-card__image-wrapper near-places__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/room.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      80
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/offer"
                    onClick={[Function]}
                  >
                    Wood and stone place 11
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Private Room
                </p>
              </div>
            </article>
            <article
              className="place-card near-places__card "
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              
              <div
                className="place-card__image-wrapper near-places__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/room.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      80
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/offer"
                    onClick={[Function]}
                  >
                    Wood and stone place 111
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Private Room
                </p>
              </div>
            </article>
          </div>
        </section>
      </div>
    </main>
  </div>,
]
`;
